File: src/tools/definitions/calculator.ts
import { z } from "zod";
import type { Tool } from "../types.js";

export const calculatorTool: Tool<
  { expression: string },
  { result: number }
> = {
  name: "calculator",
  description: "Evaluate a mathematical expression",
  parameters: z.object({
    expression: z.string().describe("Math expression to evaluate"),
  }),
  async execute({ expression }) {
    const result = Function(`"use strict"; return (${expression})`)() as number;
    return { result };
  },
};

File: src/tools/definitions/calculator.py
from src.tools.types import Tool


def _execute(expression: str) -> dict[str, float]:
    result = eval(expression)  # noqa: S307
    return {"result": float(result)}


calculator_tool = Tool(
    name="calculator",
    description="Evaluate a mathematical expression",
    parameters={
        "type": "object",
        "properties": {
            "expression": {"type": "string", "description": "Math expression to evaluate"},
        },
        "required": ["expression"],
    },
    execute=_execute,
)
